src/module.js

```diff
@@ -80,11 +87,13 @@ function module_copy(copy, injectByAlias, injectModule, map) {
       var sourceInput = source._inputs[0],
           sourceModule = sourceInput._module,
           targetModule = map.get(sourceModule) || sourceModule._copy(new Module(copy._runtime), none, null, map);
+      if (sourceModule._ready === false) ready = false;
```

Comment:
I'm still not entirely clear on why, when we observe that `sourceModule._ready` is false at this point, we can't throw the Error right here.

Reply:
This is asynchronous, so that would result in an unhandled rejection. And we can’t make module.derive return a promise (because of how it’s used in the v=3 compiled format). But there might be a way to detect this error asynchronously to cause the import-with as a whole to fail, rather than effectively making it a no-op, as in this PR currently.